ground/src/test/java/com/google/android/ground/ui/offlineareas/viewer/OfflineAreaViewerViewModelTest.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.offlineareas.viewer
+
+import com.google.android.ground.BaseHiltTest
+import com.google.android.ground.persistence.local.stores.LocalOfflineAreaStore
+import com.google.android.ground.ui.map.MapType
+import com.google.common.truth.Truth.assertThat
+import com.sharedtest.FakeData.TEST_OFFLINE_AREA
+import dagger.hilt.android.testing.HiltAndroidTest
+import javax.inject.Inject
+import kotlinx.coroutines.test.advanceUntilIdle
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@HiltAndroidTest
+@RunWith(RobolectricTestRunner::class)
+class OfflineAreaViewerViewModelTest : BaseHiltTest() {
```

Comment:
The last I talked to @sufyanAbbasi was to test more ViewModel as they contain independent logic and once all is done, we move to the fragment side.

So, in brief the plan is to have independent test which IMO is a good approach. 

WDYT?

Reply:
Also, these are not UI test as we are testing the ViewModel variables and functions, values are correct or not.

It could be UI test if we are confirming that the values are visible on the screen by opening a fragment.